src/checkpoint.hh

```diff
@@ -934,6 +1019,124 @@ SUPPORTED_EXTERNAL_TYPES = {"Rope", "answer_pknot_mfe", "pktype",
      dec_indent(); dec_indent();
   }
 
+  void archive_cyk_indices(Printer::Base &stream, size_t n_tracks) {
+     inc_indent();
+     stream << indent() << "void archive_cyk_indices() {" << endl;
+     inc_indent();
+     stream << indent() << "// save the cyk loop indidces to disk" << endl;
+     stream << indent() << "try {" << endl;
+     inc_indent();
+     stream << indent() << "std::ofstream array_fout("
+            << "tmp_out_cyk_path.c_str()"
+            << ", std::ios::binary);" << endl;
+     stream << indent() << "if (!(array_fout.good())) {" << endl;
+     stream << indent() << "  throw std::ofstream::failure(\"\");" << endl;
+     stream << indent() << "}" << endl;
+     stream << indent() << "boost::archive::binary_oarchive "
+                            "array_out(array_fout);" << endl;
+     for (size_t i = 0; i < n_tracks; i++) {
+       stream << indent() << "array_out << t_" << i << "_i + 1;" << endl;
+       stream << indent() << "array_out << t_" << i << "_j - 1;" << endl;
```

Comment:
why the +/- 1 offset? I figure to ensure that a loop iteration is not half computed. I am thinking about my outside version where indices run into the opposite direction. Can you leave a note here such that I will be reminded when merging outside?

Reply:
I added the offset, because you mentioned it might be safer if we don't continue from the exact loop index at the point of the dump but from 1 iteration prior. So I chose to go -1 for the outer loop because the index is incremented and +1 for the inner loop index because it is decremented.